lib/pages/chat/chat.dart

```diff
@@ -855,6 +855,12 @@ class ChatController extends State<ChatPageWithRoom>
         future: () async {
           if (event.status.isSent) {
             if (event.canRedact) {
+              if (room.canChangeStateEvent(EventTypes.RoomPinnedEvents)) {
+                final pinnedEvents = room.pinnedEventIds
+                    .where((e) => e != event.eventId)
+                    .toList();
+                await room.setPinnedEvents(pinnedEvents);
```

Comment:
This always updates the pinned events, and does not check if it is even in there, right? That would lead to a lot of unnecessary network requests